### DISCUSSION TITLE
How/Where do I setup TLS certificate when working with source code?

### DISCUSSION THREAD

**Discussion by r-shafi** [Keyfactor/signserver-ce#90] [Q&A]
Hello, I have cloned the [https://github.com/Keyfactor/signserver-ce](https://github.com/Keyfactor/signserver-ce) repository and setup my environment following [https://doc.primekey.com/signserver/signserver-installation](https://doc.primekey.com/signserver/signserver-installation).

I have been able to run everything properly, and configured signserver_deploy.properties to my requirements following official docs.

However, I am unable to access SignServer Admin Web because I do not have the TLS certificate set up. I have already generated the required certificates with EJBCA, now I just need to know where to place them so SignServer is able to access this.

Shafi Rayhan.

**Comment by AkshayD20** [comment]
1st you need to genrate keystore.jks and truststore.jks for accessing adminweb
then you are using Wildfly 24 or 26 put those certficate into keystore folder into 
\wildfly-preview-26.1.3.Final\standalone\configuration\keystore
add those keystore path and truststore path into standalone.xml into <tls>
<tls>
<tls>
        <key-stores>
            <key-store name="applicationKS">
                <credential-reference clear-text="password"/>
                <implementation type="JKS"/>
                <file path="application.keystore" relative-to="jboss.server.config.dir"/>
            </key-store>
            <key-store name="httpsKS">
                <credential-reference clear-text="serverpwd"/>
                <implementation type="JKS"/>
                <file path="keystore/keystore.jks" relative-to="jboss.server.config.dir"/>
            </key-store>
            <key-store name="httpsTS">
                <credential-reference clear-text="changeit"/>
                <implementation type="JKS"/>
                <file path="keystore/truststore.jks" relative-to="jboss.server.config.dir"/>
            </key-store>
        </key-stores>
        <key-managers>
            <key-manager name="applicationKM" key-store="applicationKS" generate-self-signed-certificate-host="server">
                <credential-reference clear-text="password"/>
            </key-manager>
            <key-manager name="httpsKM" algorithm="SunX509" key-store="httpsKS">
                <credential-reference clear-text="serverpwd"/>
            </key-manager>
        </key-managers>
        <trust-managers>
            <trust-manager name="httpsTM" key-store="httpsTS"/>
        </trust-managers>
        <server-ssl-contexts>
            <server-ssl-context name="applicationSSC" key-manager="applicationKM"/>
            <server-ssl-context name="httpspub" protocols="TLSv1.3" key-manager="httpsKM" trust-manager="httpsTM"/>
            <server-ssl-context name="httpspriv" protocols="TLSv1.3" key-manager="httpsKM" trust-manager="httpsTM"/>
        </server-ssl-contexts>
    </tls>
 restart wildfly and try to access adminweb

**Reply by r-shafi** [reply]
Hi Akshay, thank you for your answer, would you be able to provide me some sample code for the configurations of standalone.xml?
